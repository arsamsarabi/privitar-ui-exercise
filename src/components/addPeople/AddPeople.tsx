import React, { FC, ReactElement, useState, ChangeEvent } from "react";

import {
  StyledAddPeople,
  StyledTextArea,
  StyledButton,
} from "./StyledAddPeople";

interface IAddPeopleProps {
  handleSubmit(value: string): void;
}

export const AddPeople: FC<IAddPeopleProps> = ({
  handleSubmit,
}): ReactElement => {
  const [value, setValue] = useState<string>("");

  const handleChange = (event: ChangeEvent<HTMLTextAreaElement>): void => {
    setValue(event.target.value);
  };

  return (
    <StyledAddPeople id="addPeople">
      <StyledTextArea
        onChange={handleChange}
        placeholder="Name, Country, Age, Risk percentage"
        rows={10}
        value={value}
        id="textarea"
      />
      <StyledButton
        onClick={() => handleSubmit(value)}
        disabled={!value.length}
        id="submit-button"
      >
        SUBMIT
      </StyledButton>
    </StyledAddPeople>
  );
};
