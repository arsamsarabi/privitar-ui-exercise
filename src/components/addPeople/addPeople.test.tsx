import React from "react";
import { shallow } from "enzyme";

import {
  StyledAddPeople,
  StyledTextArea,
  StyledButton,
} from "./StyledAddPeople";
import { AddPeople } from "./AddPeople";

describe("StyledAddPeople", () => {
  it("Should render as expected", () => {
    const component = shallow(<StyledAddPeople />);
    expect(component).toMatchSnapshot();
  });
});
