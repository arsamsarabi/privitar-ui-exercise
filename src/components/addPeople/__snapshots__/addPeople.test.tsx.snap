// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledAddPeople Should render as expected 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 24px;
}

@media (min-width:768px) {
  .c0 {
    margin-top: 32px;
  }
}

<section
  className="c0"
/>
`;
